pkg/operator/endpoints/metrics.go

```diff
@@ -0,0 +1,449 @@
+				Period: aws.Int64(period),
+			},
+		},
+		{
+			Id:    aws.String("avg"),
+			Label: aws.String("Avg"),
+			MetricStat: &cloudwatch.MetricStat{
+				Metric: metric,
+				Period: aws.Int64(period),
+				Stat:   aws.String("Average"),
+			},
+		},
+	}
+
+	return regressionMetric
+}
+
+func GetNetworkStatsDef(appName string, api *context.API, period int64) []*cloudwatch.MetricDataQuery {
+	dimensions := GetAPIDimensions(appName, api)
+
+	status200 := append([]*cloudwatch.Dimension{}, dimensions...)
+	status200 = append(status200, &cloudwatch.Dimension{
+		Name:  aws.String("Code"),
+		Value: aws.String("2XX"),
+	})
+	status400 := append([]*cloudwatch.Dimension{}, dimensions...)
+	status400 = append(status400, &cloudwatch.Dimension{
+		Name:  aws.String("Code"),
+		Value: aws.String("4XX"),
+	})
+	status500 := append([]*cloudwatch.Dimension{}, dimensions...)
+	status500 = append(status500, &cloudwatch.Dimension{
+		Name:  aws.String("Code"),
+		Value: aws.String("5XX"),
+	})
+
+	networkDataQueries := []*cloudwatch.MetricDataQuery{
+		{
+			Id:    aws.String("datapoints_2XX"),
+			Label: aws.String("2XX"),
+			MetricStat: &cloudwatch.MetricStat{
+				Metric: &cloudwatch.Metric{
+					Namespace:  aws.String(config.Cortex.Namespace),
+					MetricName: aws.String("StatusCode"),
+					Dimensions: status200,
+				},
+				Stat:   aws.String("Sum"),
+				Period: aws.Int64(period),
+			},
+		},
+		{
+			Id:    aws.String("datapoints_4XX"),
+			Label: aws.String("4XX"),
+			MetricStat: &cloudwatch.MetricStat{
+				Metric: &cloudwatch.Metric{
+					Namespace:  aws.String(config.Cortex.Namespace),
+					MetricName: aws.String("StatusCode"),
+					Dimensions: status400,
+				},
+				Stat:   aws.String("Sum"),
+				Period: aws.Int64(period),
+			},
+		},
+		{
+			Id:    aws.String("datapoints_5XX"),
+			Label: aws.String("5XX"),
+			MetricStat: &cloudwatch.MetricStat{
+				Metric: &cloudwatch.Metric{
+					Namespace:  aws.String(config.Cortex.Namespace),
+					MetricName: aws.String("StatusCode"),
+					Dimensions: status500,
+				},
+				Period: aws.Int64(period),
+				Stat:   aws.String("Sum"),
+			},
+		},
+	}
+
+	return networkDataQueries
+}
+
+func GetClassesMetricDef(appName string, api *context.API, period int64) ([]*cloudwatch.MetricDataQuery, error) {
+	classMetricQueries := []*cloudwatch.MetricDataQuery{}
+	listMetricsInput := &cloudwatch.ListMetricsInput{
+		Namespace:  aws.String(config.Cortex.Namespace),
+		MetricName: aws.String("Prediction"),
+		Dimensions: []*cloudwatch.DimensionFilter{
+			{
+				Name:  aws.String("AppName"),
+				Value: aws.String(appName),
+			},
+			{
+				Name:  aws.String("APIName"),
+				Value: aws.String(api.Name),
+			},
+			{
+				Name:  aws.String("APIID"),
+				Value: aws.String(api.ID),
+			},
+		},
+	}
+
+	listMetricsOutput, err := config.AWS.CloudWatchMetrics.ListMetrics(listMetricsInput)
+	if err != nil {
+		return nil, err
+	}
+
+	if listMetricsOutput.Metrics == nil {
+		return classMetricQueries, nil
```

Comment:
nit: when returning blank or nil values, it's more clear to be explicit (e.g. `return nil, nil` here)